### PR TITLE
CASMINST-4963 Network upgrade and CAN-to-CHN procedure

### PR DESCRIPTION
**Pull request by trad511**
# Description

- The CSM 1.2 to CSM 1.3 management network upgrade procedure is contained here and has been moved from the main upgrade path.
- A procedure has been added to convert CAN networking to CHN if the system requires.  The procedure is designed to be run out-of-band after a CSM 1.3 upgrade, but may also be run during the CSM 1.3 upgrade process itself.  This procedure employs a "hot-hot" failover from the CSM 1.2 CAN to the CSM 1.3 CHN to allow maximum flexibility and to minimize outages to UAN, UAI and compute resources.  The procedure is applied in a "hot fix" fashion and does not require cluster outages, NCN reboots or service-effecting restarts.

# Checklist Before Merging

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Comment by mharding-hpe** [comment]
This is still a draft but I'm adding the "do not merge" label because I want to review this before we merge it (and have had bad luck with things being merged when I have only commented on this in my feedback but not added the label). This does not imply anything about my feelings as to the quality of the content here -- it just means that I don't know if I will have time to review this before the first time it will be considered for merging.

I will remove the label once the PR is marked ready for review, I've finished my reviews, and issues identified (if any) are addressed.

File: upgrade/network/bican_enable.md
Comment (gbaker-hpe) [inline upgrade/network/bican_enable.md]:
```suggestion
1. (`ncn-m001#`) Create a working directory for the procedure.
```

File: upgrade/network/bican_enable.md
Comment (gbaker-hpe) [inline upgrade/network/bican_enable.md]:
```suggestion
2. (`ncn-m001#`) Obtain an API token
```

File: upgrade/network/bican_enable.md
Comment (gbaker-hpe) [inline upgrade/network/bican_enable.md:59]: Are there any automated tests that should be run to determine this?

File: upgrade/network/bican_enable.md
Comment (gbaker-hpe) [inline upgrade/network/bican_enable.md:142]: Do we?

File: upgrade/network/bican_enable.md
Comment (gbaker-hpe) [inline upgrade/network/bican_enable.md:202]: I think what we want to say here is

* Be really sure you have the latest 1.3 docs because the scripts we're about to call are shipped with those docs
* Set this env var which points to the sls scripts included with the docs
* Now proceed to do these things that run those scripts

File: upgrade/network/bican_enable.md
Comment (gbaker-hpe) [inline upgrade/network/bican_enable.md:96]: Let's recommend that they save these files off to an off-system VCS repo (assuming they do not contain secrets).

File: upgrade/network/bican_enable.md
Comment (gbaker-hpe) [inline upgrade/network/bican_enable.md:411]: TODO!

File: upgrade/network/bican_enable.md
Comment (trad511) [reply upgrade/network/bican_enable.md]: the shadow knows...?

File: upgrade/network/bican_enable.md
Comment (trad511) [reply upgrade/network/bican_enable.md]: I've put docs requirements and links up in prerequisites and removed this and other shenanigans like this.

File: upgrade/network/bican_enable.md
Comment (trad511) [reply upgrade/network/bican_enable.md]: We've performed the update and it works.  Attempting to have more automation than a series of "vi this and cut-n-paste that" which makes me sad.

File: upgrade/network/bican_enable.md
Comment (trad511) [reply upgrade/network/bican_enable.md]: **Recommend** copying and storing all data in ${BACKUPDIR} off-system in a version control repository is hightly recommended. has been added.  similar statements for UPDATE AND CLEANUP phases as well.

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:74]: Also need an IP for oath2-proxy service

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:106]: Would be great if we could combine all of this into a single script.

**Comment by trad511** [comment]
@johren-hpe @spillerc-hpe @gbaker-hpe still keeping this in draft PR but would like a review from you.  Luke and I ran through the procedure (all except compute to CHN) on Surtur and looks good to us.  Will implement any changes you request then rebase and get the PR out there proper.

File: upgrade/network/chn_enable.md
Comment (spillerc-hpe) [inline upgrade/network/chn_enable.md:264]: In the Papaya recipe CFS is no longer responsible for configuring the HSN interfaces, the SHS software is baked into the NCN image and the `cray-ifscan` systemd service performs configuration of the HSN interfaces.

I wonder if there is any SHS/Slingshot troubleshooting documentation for this that we can point users to?

File: upgrade/network/network_upgrade_1.2_to_1.3.md
Comment (spillerc-hpe) [inline upgrade/network/network_upgrade_1.2_to_1.3.md:122]: Should the version here be 1.3?

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:197]: Do we want to give them the value that we usually use for this vlan as an example?

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:451]: Should active be activate?

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:489]: "and the management switches"?

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:734]: s/to/from ?

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:738]: What if CHN was added for computes above?   It think this will end up removing those CHN IPs from SLS, right?

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:756]: typo:  remoe -> remove

File: upgrade/network/chn_enable.md
Comment (johren-hpe) [inline upgrade/network/chn_enable.md:461]: typo:  durig -> during

File: upgrade/network/network_upgrade_1.2_to_1.3.md
Comment (johren-hpe) [inline upgrade/network/network_upgrade_1.2_to_1.3.md:20]: Is it CMN or CAN vlan that is being removed from UAN?

File: upgrade/network/network_upgrade_1.2_to_1.3.md
Comment (johren-hpe) [inline upgrade/network/network_upgrade_1.2_to_1.3.md:60]: What if they are doing this CAN->CHN migration after a 1.3 upgrade?   Will the 1.2 folder be confusing?

File: upgrade/network/network_upgrade_1.2_to_1.3.md
Comment (johren-hpe) [inline upgrade/network/network_upgrade_1.2_to_1.3.md:114]: What if we release a newer version of CANU along the way?

File: upgrade/network/chn_enable.md
Comment (trad511) [reply upgrade/network/chn_enable.md:197]: This has just lead to problems in the past so I'd prefer not to do this.  Plus defaults are baked into the python scripts and available on error or `--help`.

File: upgrade/network/network_upgrade_1.2_to_1.3.md
Comment (trad511) [reply upgrade/network/network_upgrade_1.2_to_1.3.md:20]: This is actually the CMN in this case.  In CSM 1.2 upgrade we created CAN and moved traffic to CAN but didn't lock out the CMN from UAN until CSM 1.3.  Reads weird but true.  I'll add some more text.

File: upgrade/network/bican_enable.md
Comment (trad511) [reply upgrade/network/bican_enable.md]: not needed.

File: upgrade/network/chn_enable.md
Comment (trad511) [reply upgrade/network/chn_enable.md:108]: Given the limited audience and one-time nature of this I would prefer not to wrap this in a script.

File: upgrade/network/chn_enable.md
Comment (trad511) [reply upgrade/network/chn_enable.md:160]: Given the limited audience and one-time nature of this I would prefer not to wrap this in a script.

File: upgrade/network/chn_enable.md
Comment (trad511) [reply upgrade/network/chn_enable.md:738]: yup. good find.

File: upgrade/network/network_upgrade_1.2_to_1.3.md
Comment (trad511) [reply upgrade/network/network_upgrade_1.2_to_1.3.md:60]: This is the generalized 1.2 to 1.3 upgrade procedure.  The use cases I see are:
1. 1.3 fresh install - will never get to this doc, nor the can->chn doc and switches will be configured early as normal.
2. 1.3 upgrade can->can will never hit the can->chn doc and the upgrade will end the switch updates for this release.
3. 1.3 upgrade chn->chn will never hit the can->chn doc and the upgrade will end the switch updates for this release.
4. 1.3 upgrade can->chn during upgrade will hit the chn_enable doc and then be passed here as the last step.
5. 1.3 upgrade complete.. then apply chn_enable procedure. they would upgrade can->can and then run the procedure so see item 2.  

will put some verbiage in to cover use case 5.

File: upgrade/network/chn_enable.md
Comment (trad511) [reply upgrade/network/chn_enable.md:12]: we've decided to not do this step for now. these records would be used "only if unbound blips" and only useful to workers.

**Comment by rustydb** [comment]
Rebased and squashed the commits to help clear up the diff.
Used `--force-with-lease`, leaving the old commits present on GitHub.

**Review by lukebates123 — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Review by keopp-hpe — APPROVED** [review]

**Review by gbaker-hpe — APPROVED** [review]

**Comment by gbaker-hpe** [comment]
Merged this in to avoid keeping this branch so long-lived.  We can follow up with additional changes or fixes in separate PRs.

**Pull request merged**
